hbw/app/javascript/packs/hbw/components/form/services_table/service_add_modal.tsx

```diff
@@ -0,0 +1,332 @@
+                        <span className='hbw-datetime-label'>
+                          {translate('components.services_table.modals.service_add.end_date')}
+                        </span>
+                        <DatePickerWrapper
+                          aria-label='end-date'
+                          name='end-date-visible-input'
+                          className='form-control'
+                          onChange={handleEndDateChange}
+                          selected={endDate}
+                          dateFormat={dateFormat}
+                          locale={locale.code}
+                          fixedHeight
+                          disabledKeyboardNavigation
+                          disabled={false}
+                          popperPlacement='bottom-start'
+                          autoComplete='off'
+                          showTimeInput={false}
+                        />
+                      </div>
+                    </div>
+                  </div>
+                  <div className='row'>
+                    <div className='col-xs-12 col-sm-12 col-md-12 col-lg-12'>
+                      <div className='form-group'>
+                        <label className='hbw-checkbox-label'>
+                          <input
+                            name='allow-provisioning'
+                            type='checkbox'
+                            className='hbw-checkbox'
+                            id='allow-provisioning'
+                            checked={allowProvisioning}
+                            onChange={handleAllowProvisioningChange}
+                          />
+                          <FontAwesomeIcon
+                            className='hbw-checkbox'
+                            icon={checkboxIcon(allowProvisioning) as IconProp}
+                          />
+                          <span>
+                            {` ${translate('components.services_table.modals.service_add.allow_provisioning')}`}
+                          </span>
+                        </label>
+                      </div>
+                    </div>
+                  </div>
+                </div>
+              </div>
+              <div className='modal-footer footer'>
+                <button
+                  type='button'
+                  className='btn btn-primary close-add-service-button'
+                  data-dismiss='modal'
+                  onClick={handleClose}>
+                  {translate('components.services_table.modals.service_add.cancel')}
+                </button>
+                <button
+                  type='submit'
+                  className='btn btn-info submit-add-service-button'
+                  data-dismiss='modal'>
+                  {translate('components.services_table.modals.service_add.submit')}
+                </button>
+              </div>
+            </form>
+          </div>
+        </div>
+      </div>
+    </div>
+  </div>, document.body);
```

Comment:
are you okay with this 200-lines jsx?
I'm not, at least a modal itself could be a separate abstract wrapper